samples/dnn/dasiamrpn_tracker.py

```diff
@@ -14,38 +18,62 @@
 import sys
 
 class DaSiamRPNTracker:
-    #initialization of used values, initial bounding box, used network
-    def __init__(self, im, target_pos, target_sz, net, kernel_r1, kernel_cls1):
+    # Initialization of used values, initial bounding box, used network
+    def __init__(self):
         self.windowing = "cosine"
         self.exemplar_size = 127
         self.instance_size = 271
         self.total_stride = 8
-        self.score_size = (self.instance_size - self.exemplar_size) // self.total_stride + 1
+        self.score_size = (
+            self.instance_size - self.exemplar_size) // self.total_stride + 1
         self.context_amount = 0.5
         self.ratios = [0.33, 0.5, 1, 2, 3]
         self.scales = [8, ]
         self.anchor_num = len(self.ratios) * len(self.scales)
         self.penalty_k = 0.055
         self.window_influence = 0.42
         self.lr = 0.295
+        self.score = []
+        if self.windowing == "cosine":
+            self.window = np.outer(
+                np.hanning(self.score_size), np.hanning(self.score_size))
+        elif self.windowing == "uniform":
+            self.window = np.ones((self.score_size, self.score_size))
+        self.window = np.tile(self.window.flatten(), self.anchor_num)
+
+    def init(self, im, init_bb, net="dasiamrpn_model.onnx", \
+        kernel_r1="dasiamrpn_kernel_r1.onnx", \
+        kernel_cls1="dasiamrpn_kernel_cls1.onnx"):
+        target_pos, target_sz = np.array([init_bb[0], init_bb[1]]), np.array(
+            [init_bb[2], init_bb[3]])
         self.im_h = im.shape[0]
         self.im_w = im.shape[1]
         self.target_pos = target_pos
         self.target_sz = target_sz
         self.avg_chans = np.mean(im, axis=(0, 1))
-        self.net = net
-        self.score = []
-
-        if ((self.target_sz[0] * self.target_sz[1]) / float(self.im_h * self.im_w)) < 0.004:
-             raise AssertionError("Initializing BB is too small-try to restart tracker with larger BB")
+        # Loading network`s and kernel`s models
+        net = cv.dnn.readNet(net)
+        kernel_r1 = cv.dnn.readNet(kernel_r1)
+        kernel_cls1 = cv.dnn.readNet(kernel_cls1)
```

Comment:
Is that possible to reinitialize a tracker without model reloading?

Reply:
I tried to load models once, and as I understand, it is worked with [the LaSOT benchmark](https://github.com/opencv/opencv_contrib/pull/2516).

Reply:
So, please move model loading into constructor (with model paths arguments).